core/services/ping/pingmanager.py

```diff
@@ -27,15 +27,27 @@ async def register_ethernet_ping360(self, ping: PingDeviceDescriptor) -> None:
         if ping not in self.drivers:
             self.drivers[ping] = Ping360Driver(ping, -1)
 
+    def find_next_port(self, base_port: int, direction: int) -> int:
+        """
+        Find the next unused udp port, starts on base_port and increments/decrements
+        accordingly to direction
+        """
+        port = base_port
+        while udp_port_is_in_use(port):
```

Comment:
We should add a sleep here to avoid cpu lock if the `udp_port_is_un_use` always return a true somehow.